src/app/beer_garden/api/stomp/server.py

```diff
@@ -0,0 +1,150 @@
+import sys
+import stomp
+import logging
+import time
+import beer_garden.config as config
+from brewtils.models import Event, Events, Request, Operation, System
+from brewtils.schema_parser import SchemaParser
+import beer_garden.events
+import beer_garden.router
+from beer_garden.api.stomp.processors import append_headers, process_send_message
+
+
+conn = None
+bg_active = False
+
+
+def send_message(message, headers={}):
+    global conn
+    stomp_config = config.get("entry.stomp")
+    message, response_headers = process_send_message(message)
+    response_headers = append_headers(
+        response_headers=response_headers, request_headers=headers
+    )
+    if conn.is_connected():
+        if "reply-to" in headers:
+            conn.send(
+                body=message, headers=response_headers, destination=headers["reply-to"]
+            )
+        else:
+            conn.send(
+                body=message,
+                headers=response_headers,
+                destination=stomp_config.event_destination,
+            )
+
+
+def send_error_msg(error, headers):
+    global conn
+    stomp_config = config.get("entry.stomp")
+    error_headers = None
+    error_headers = append_headers(error_headers, headers)
+
+    if conn.is_connected():
+        if "reply-to" in headers:
+            conn.send(
+                body=error.__str__(),
+                headers=error_headers,
+                destination=headers["reply-to"],
+            )
+        else:
+            conn.send(
+                body=error.__str__(),
+                headers=error_headers,
+                destination=stomp_config.event_destination,
+            )
+    pass
+
+
+class OperationListener(stomp.ConnectionListener):
+    def on_error(self, headers, message):
+        print("received an error:", headers)
+
+    def on_message(self, headers, message):
+        error = None
+        error_msg = None
+        operation = None
+        try:
+            operation = SchemaParser.parse_operation(message, from_string=True)
+            if hasattr(operation, "kwargs"):
+                operation.kwargs.pop("wait_timeout", None)
+        except:
+            error_msg = "Failed to parse message"
+            error = sys.exc_info()
+
+        result = None
+        if error_msg is None:
+            try:
+                result = beer_garden.router.route(operation)
+            except:
+                error_msg = "Failed to route operation"
+                error = sys.exc_info()
```

Comment:
We have a bunch of different exceptions that couple be raised during this. We should properly reflect which exception was raised.We raise the exception So you can just capture the error message and return the message.

``` 
except Exception as e:
    error_msg = str(e)
```